File: packages/kira_live/src/platform.zig
const std = @import("std");
const RunnerKind = @import("runner_kind.zig").RunnerKind;

pub const LivePlatform = enum {
    desktop,
    ios_simulator,
    ios_device,

    pub fn parse(text: []const u8) ?LivePlatform {
        if (std.mem.eql(u8, text, "desktop")) return .desktop;
        if (std.mem.eql(u8, text, "ios") or
            std.mem.eql(u8, text, "ios-simulator") or
            std.mem.eql(u8, text, "simulator"))
        {
            return .ios_simulator;
        }
        if (std.mem.eql(u8, text, "ios-device") or
            std.mem.eql(u8, text, "device"))
        {
            return .ios_device;
        }
        return null;
    }

    pub fn cliName(self: LivePlatform) []const u8 {
        return switch (self) {
            .desktop => "desktop",
            .ios_simulator => "ios-simulator",
            .ios_device => "ios-device",
        };
    }

    pub fn runnerKind(self: LivePlatform) ?RunnerKind {
        return switch (self) {
            .desktop => .desktop_dynamic_host,
            .ios_simulator, .ios_device => null,
        };
    }
};

test "LivePlatform parses user-facing aliases" {
    try std.testing.expectEqual(LivePlatform.desktop, LivePlatform.parse("desktop").?);
    try std.testing.expectEqual(LivePlatform.ios_simulator, LivePlatform.parse("ios").?);
    try std.testing.expectEqual(LivePlatform.ios_simulator, LivePlatform.parse("ios-simulator").?);
    try std.testing.expectEqual(LivePlatform.ios_device, LivePlatform.parse("ios-device").?);
}

File: crates/kira-live/src/platform.rs
//! Live platform identity: maps manifest runner ids to concrete runners.
//!
//! Ported from kira-zig `kira_live/src/platform.zig`.

use crate::runner_kind::RunnerKind;

pub use kira_manifest::RunnerId;

/// The live platform id (same domain as the manifest runner id).
pub type LivePlatform = RunnerId;

/// Parse a user-facing runner id (including aliases like `ios-simulator`).
pub fn parse_runner_id(text: &str) -> Option<RunnerId> {
    RunnerId::parse(text)
}

/// The concrete runner behind a platform id.
pub fn runner_kind(id: RunnerId) -> RunnerKind {
    match id {
        RunnerId::Desktop => RunnerKind::DesktopDynamicHost,
        RunnerId::Macos => RunnerKind::XcodeMacos,
        RunnerId::Ios => RunnerKind::XcodeIos,
        RunnerId::Tvos => RunnerKind::XcodeTvos,
        RunnerId::Visionos => RunnerKind::XcodeVisionos,
        RunnerId::Windows => RunnerKind::WindowsVisualStudio,
        RunnerId::Android => RunnerKind::AndroidGradle,
        RunnerId::Web => RunnerKind::WebKiraWasm,
        RunnerId::Linux => RunnerKind::LinuxCmake,
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn parses_user_facing_aliases() {
        assert_eq!(Some(RunnerId::Desktop), parse_runner_id("desktop"));
        assert_eq!(Some(RunnerId::Ios), parse_runner_id("ios-simulator"));
        assert_eq!(RunnerKind::WebKiraWasm, runner_kind(RunnerId::Web));
    }
}
